Backend/src/main/java/com/example/appengine/springboot/SpringbootApplication.java

```diff
@@ -40,18 +35,37 @@ public static void main(String[] args) {
         SpringApplication.run(SpringbootApplication.class, args);
     }
 
+
     // Controls Searches of all data Sources
     @GetMapping("/score/{company}")
     public Business masterController(@PathVariable("company") String companyName) throws IOException {
-        Business business = new Business();
-        business.update(isBlueSignPartner(companyName));
-        business.update(searchBCorp(companyName));
+        int searchWordMax = 3;
+        companyName = prepareSearchTerm(companyName);
+        String companyNameArray [] = companyName.split("_");
+        String searchWordsArray [] =
+                new String[((companyNameArray.length < searchWordMax) ? companyNameArray.length : searchWordMax)];
+        for (int a = 0; a < searchWordsArray.length; ++a) {
+            searchWordsArray[a] = companyNameArray[a];
+        }Business business = new Business();
+        String searchWord = "";
+        for (int a = searchWordsArray.length; a > 0; --a) {
+            searchWord = searchWordsArray[0];
```

Comment:
```suggestion

        for (int a = searchWordsArray.length; a > 0; --a) {
            String searchWord = searchWordsArray[0];
```
It is cleaner to do it like this and since Java Strings are immutable, you aren't saving anything by doing it the other way